src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java

```diff
@@ -279,5 +302,68 @@ public boolean delete() {
         // TODO: Add journey state deletion.
         return Persistence.monitoredTrips.removeById(this.id);
     }
+
+    public List<NameValuePair> getParsedParams() throws URISyntaxException {
+        return URLEncodedUtils.parse(
+            new URI(String.format("http://example.com/%s", queryParams)),
+            UTF_8
+        );
+    }
+
+    public boolean isArriveBy() throws URISyntaxException {
+        for (NameValuePair param : getParsedParams()) {
+            if (param.getName().equals("arriveBy")) {
+                return param.getValue().equals("true");
+            }
+        }
+
+        // if arriveBy is not included in query params, OTP will default to false, so initialize to false
+        return false;
+    }
+
+    /**
+     * Gets the timezone of the target location. When planning a trip, we either want to depart at a certain time or 
+     * arrive by a certain time. When departing, we use the local time present at the origin and when arriving by we 
+     * use the local time at the destination. Therefore, this method will return the timezone at the destination if this
+     * trip is an arriveBy trip, or the timezone at the origin if the trip is a depart at trip.
+     */
+    @BsonIgnore
```

Comment:
Same here. `@BsonIgnore` and `JsonIgnore` are... ignored by the Mongo POJO codec on serialization. The easiest way is to rename the method (`findTimezoneForTargetLocation`?). If a `get` prefix is still needed, we can use the `transient` keyword ([we've faced a similar issue here](https://github.com/ibi-group/otp-middleware/commit/a5baa3c85c6741aac58473e2110ab206221aa993))